simplq/src/components/pages/Status/LeaveQueue.jsx

```diff
@@ -1,16 +1,22 @@
 import React from 'react';
+import CloseIcon from '@material-ui/icons/Close';
+import IconButton from '@material-ui/core/IconButton';
+import { useHistory } from 'react-router-dom';
 import styles from '../../../styles/statusPage.module.scss';
 
 export default () => {
+  const history = useHistory();
+
   return (
-    <>
-      <div className={styles['user-action']}>
-        {/* icon */}
-        <div>
-          <h2>Leave Queue</h2>
-          <p>Exit from the queue</p>
-        </div>
+    <li className={styles['user-action']}>
+      <IconButton onClick={() => history.push(`/`)} aria-label="Leave Queue">
+        <CloseIcon />
```

Comment:
Could you add a props.leaveQueueHandler call here? It is passed in to the sidePanel right now.

  ` <StatusSidePanel leaveQueueHandler={onDeleteClick} queueId={tokenStatusResponse.queueId} />`

You could pass this further into LeaveQueue and call it along with the history.push and I believe that should work.

Reply:
I never though of this, but taking the user to home page after `props.leaveQueueHandler` is a really good idea!

Reply:
Done